AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/script_canvas_tools_qt.py

```diff
@@ -0,0 +1,590 @@
+    a script canvas graph and node inspector qt widget.
+
+    param self: the script calling this function
+    param sc_graph_node_inspector: the sc graph node inspector qt widget
+    param sc_graph: the sc graph qt widget
+
+    returns: a list of node titles (i.e Print - Utilities/Debug). If there are duplicates of a node then the title
+    will include ( X Selected) in the string.
+    """
+    node_inspector_scroll_area = sc_graph_node_inspector.findChild(QtWidgets.QScrollArea, "")
+    # perform ctrl+a keystroke to highlight all nodes on the graph
+    QtTest.QTest.keyClick(sc_graph, "a", Qt.ControlModifier, WAIT_FRAMES)
+    node_inspector_backgrounds = node_inspector_scroll_area.findChildren(QtWidgets.QFrame, "Background")
+    titles = []
+    for background in node_inspector_backgrounds:
+        background_title = background.findChild(QtWidgets.QLabel, NODE_INSPECTOR_TITLE_KEY)
+        if background_title.text() != "":
+            titles.append(background_title.text())
+    return titles
+
+
+def get_main_sc_window_qt_object():
+    """
+    function for getting the sc main window qt object.
+
+    params: none
+
+    returns: a qt widget main window object
+    """
+    editor_window = pyside_utils.get_editor_main_window()
+    sc_editor = editor_window.findChild(QtWidgets.QDockWidget, SCRIPT_CANVAS_UI)
+    return sc_editor.findChild(QtWidgets.QMainWindow)
+
+
+def create_new_sc_graph():
+    """
+    function for opening a new script canvas graph file. uses the sc editor window to trigger a new file action
+
+    param self: the script calling this function
+    param sc_editor_main_window: the qt object for the main sc_editor window
+
+    returns: none
+    """
+    sc_editor_main_window = EDITOR_QT_CONTAINER.get_SC_editor_wrapper().get_sc_main_pane()
+    create_new_graph_action = pyside_utils.find_child_by_pattern(
+        sc_editor_main_window, {"objectName": "action_New_Script", "type": QtWidgets.QAction}
+    )
+    create_new_graph_action.trigger()
+
+
+def create_new_SC_variable(new_variable_type):
+    """
+    function for creating a new SC variable through variable manager qt object
+
+    param variable_type: The variable data type to create as a string. i.e "Boolean"
+    returns: none
+    """
+
+    variable_manager = EDITOR_QT_CONTAINER.get_variable_manager()
+    variable_manager.create_new_variable(new_variable_type)
+
+
+def verify_SC_variable_count(expected):
+    """
+    function to check if the current number of variables in variable manager matches the user provided input
+
+    param expected: the expected number of variables in the variable manager
+
+    returns true if the actual number of variables in the variable manager matches the expected number
+    """
+    variable_manager = EDITOR_QT_CONTAINER.get_variable_manager()
+    row_count = variable_manager.get_variable_count()
+
+    return expected == row_count
```

Comment:
This appears to be refactored already and can be removed.

Reply:
the function on the variable manager just gets the current number of variables. this function compares that to the expected value

Reply:
In that case let's refactor this into `QtPyScriptCanvasVariableManager`.